### ISSUE TITLE
Fix Windows edge cases in unit tests

### ISSUE THREAD

**Issue by jdblischak** [workflowr/workflowr#76]
Some of the unit tests that pass on Linux and macOS fail on Windows. First reported by @vanAmsterdam  in #73:

```
Running 'testthat.R' [247s]
Running the tests in 'tests/testthat.R' failed.
Last 13 lines of output:
Running tests in test-wflow_update.R
testthat results ================================================================
OK: 596 SKIPPED: 1 FAILED: 9

    Failure: obtain_existing_paths works with relative path to home directory (@test-obtain_existing_path.R#51)
    Failure: obtain_existing_paths works with relative path to home directory (@test-obtain_existing_path.R#52)
    Failure: obtain_existing_paths works with relative path to home directory (@test-obtain_existing_path.R#53)
    Failure: obtain_existing_paths works with relative path to home directory (@test-obtain_existing_path.R#54)
    Failure: obtain_existing_paths works with relative path to home directory (@test-obtain_existing_path.R#55)
    Failure: verbose = FALSE suppresses all output (@test-wflow_convert.R#141)
    Failure: verbose = FALSE suppresses all output (@test-wflow_convert.R#144)
    Failure: wflow_start throws an error if user.name and user.email are not set (@test-wflow_start.R#190)
    Failure: wflow_start throws an error if user.name and user.email are not set (@test-wflow_start.R#192)
```

**Comment by jdblischak** [comment]
The original report was run with R 3.4.1 on Windows 7. I was able to duplicate a subset of the errors running R 3.4.1 on Windows 10 (all except the `obtain_existing_paths` errors).

**Comment by jdblischak** [comment]
I suspect part of the issue is that `normalizePath("~")` points to the user's `Documents/` directory, and thus won't find files like `.gitconfig` in the user's home directory.

**Comment by pcarbo** [comment]
@jdblischak Is [this info](https://cran.r-project.org/bin/windows/base/rw-FAQ.html#What-are-HOME-and-working-directories_003f) useful?

**Comment by jdblischak** [comment]
@pcarbo Yes, that is helpful. Now that I know that R considering `~` to be `~/Documents` is the default, desired behavior on Windows, I can work around it. I'll have to figure out a way to strip `Documents/`on Windows only because Git still puts `.gitconfig` in the user's actual home directory.

**Comment by jdblischak** [comment]
Just figured out the error from wflow_convert. The unexpected warning was coming from `wflow_convert()` -> `diff_file()` -> `tools::Rdiff()`-> `diff` command-line utility. According to the man page for GNU diff:

> Exit status is 0 if inputs are the same, 1 if different, 2 if trouble.

This wasn't an issue on macOS and Linux, but Windows sends a warning when the exit status is 1 (i.e. the files have differences). I suppressed the warning since it isn't informative.

**Comment by jdblischak** [comment]
These errors have been fixed in version 0.9.0.

* `wflow_convert()`: https://github.com/jdblischak/workflowr/commit/c4acdcb8e13d1b8ec69a55445fe1bba2b68e0767
* `obtain_existing_path()`: https://github.com/jdblischak/workflowr/commit/28cbbd7843587227e6936ea9ac7b1f61ec7b5a05
* `wflow_git_config()`: https://github.com/jdblischak/workflowr/commit/23530f868081810a58fa49944e8f9d9a7a1f29ba

And I setup CI for Windows with AppVeyor, so these types of cross-platform issues should be caught more quickly in the future.
